galpy/potential_src/Potential.py

```diff
@@ -1510,6 +1510,166 @@ def nemo_accpars(self,vo,ro):
+            
+            2018-02-12 - Written - Webb (UofT)
+            
+        """
+        if self.isNonAxi:
+            #Make sure potential is axisymmetric
+            print("Error - cannot calculate rtide for non-axisymmetric potential")
+            return 0.0
+        elif M==None:
+            #If units not given, assume galpy! GM units?
+            print("Error - mass parameter has not been set")
+            return 0.0
+        else:
+            #Check mass units (default: GALPY units)
+            if M_units=='mass':
+                M/=bovy_conversion.mass_in_msol(self._vo,self._ro)
+            elif M_units=='GM':
+                M/=(bovy_conversion.mass_in_msol(self._vo,self._ro)*bovy_conversion._G)
+            # Parse M if it has units
+            elif _APY_LOADED and isinstance(M,units.Quantity):
+                # Try a bunch of possible units
+                unitFound= False
+                # mass
+                try:
+                    M= M.to(units.Msun).value\
+                        /bovy_conversion.mass_in_msol(self._vo,self._ro)
+                except units.UnitConversionError: pass
+                else:
+                    unitFound= True
+                    if not M_units == 'mass':
+                        raise units.UnitConversionError('M= parameter of %s should have units of %s, but has units of mass instead' % (type(self).__name__,M_units))
+                if not unitFound:
+                    # G x mass
+                    try:
+                        M= M.to(units.pc*units.km**2/units.s**2)\
+                            .value\
+                            /bovy_conversion.mass_in_msol(self._vo,self._ro)\
+                            /bovy_conversion._G
+                    except units.UnitConversionError: pass
+                    else:
+                        unitFound= True
+                        if not M_units == 'mass':
+                            raise units.UnitConversionError('M= parameter of %s should have units of %s, but has units of G x mass instead' % (type(self).__name__,M_units))
+                                    
+            #To calculate omegac in spherical coordinates we need the first derivative of the potential with respect to r
+            r=math.sqrt(R**2.0+z**2.0)
+            theta=nu.arctan2(R, z)
+            
+            Rderiv=-1.0*self.Rforce(R,z,use_physical=False)
+            zderiv=-1.0*self.zforce(R,z,use_physical=False)
+
+            omegac2=(Rderiv*(R/r)+zderiv*(z/r))/r
+            
+            #To calculate the epicyclic frequency kappa we need the second derivative of the potential with respect to r
+            R2deriv= self.R2deriv(R,z,use_physical=False)
+            z2deriv=self.z2deriv(R,z,use_physical=False)
+            Rzderiv=self.Rzderiv(R,z,use_physical=False)
+            
+            r2deriv=(R2deriv*nu.sin(theta)+Rzderiv*nu.cos(theta))*nu.sin(theta)+(Rzderiv*nu.sin(theta)+z2deriv*nu.cos(theta))*nu.cos(theta)
```

Comment:
I think it would be good to introduce a new method ``r2deriv`` (like the way ``rforce`` is implemented) that gives the second derivative wrt the spherical radius and then use that here. This function might be useful more generally.